Chess/common_types.h

```diff
@@ -0,0 +1,61 @@
+#ifndef COMMON_TYPES_H
+#define COMMON_TYPES_H
+
+#include <QObject>
+#include <QString>
+
+
+class Figure;
+class Desk;
+class Command;
+
+
+enum FigType{EMPTY/*base class*/,
+             KING,
+             QUEEN,
+             ROOK,
+             BISHOP,
+             KNIGHT,
+             PAWN};
+
+
+enum FigColor{NONE/*base class*/,
+              WHITE,
+              BLACK};
+
+
+struct Cell {
+    int row_;
+    int col_;
+
+    Cell(int row=-1, int col=-1):
+        row_(row),
+        col_(col)
+    {
+    }
+
+    bool operator ==(Cell rv) {
+        return (row_==rv.row_ && col_==rv.col_);
+    }
+
+};
+
+struct CellInfo {
```

Comment:
Why do we have Cell and CellInfo as separate classes?
Amd yes. make it class, not a struct